src/System.Management.Automation/engine/regex.cs

```diff
@@ -281,6 +281,55 @@ public static bool ContainsWildcardCharacters(string pattern)
             return result;
         }
 
+        /// <summary>
+        /// Checks to see if the given string may be wildcard pattern.
+        /// </summary>
+        /// <param name="pattern">
+        /// String which needs to be checked if it may be wildcard pattern
+        /// </param>
+        /// <returns> true if the string may be wildcard pattern, false otherwise. </returns>
+        /// <remarks>
+        /// Currently string contains { '*', '?' } or both { '[', ']' } is
+        /// considered to be possibly a wildcard pattern and
+        /// '`' is the escape character.
+        /// </remarks>
+        public static bool MayBeWildcardPattern(string pattern)
```

Comment:
Shouldn't we enhance `ContainsWildcardCharacters` method? 
Seem it would be better. And I'd consider using ReadOnlySpan.

Reply:
I believe `test if a string contains wildcard characters` and `test if a string looks like a wildcard pattern` is two different thing.
I can enhance `ContainsWildcardCharacters`, but I am not sure whether it will break other components...

Reply:
Yes, it can affect many things but I think it is worth it to take a deeper - split the code just because we have not learned it enough is a bad practice.
I believe that if `ContainsWildcardCharacters` method takes into account _escapes_, then it should do the same for square brackets.

Reply:
I tried to merge them, but I am not sure whether it is good enough.